bin/varnishd/cache/cache_vrt_var.c

```diff
@@ -880,22 +880,17 @@ VRT_r_resp_do_esi(VRT_CTX)
 
 #define VRT_BODY_L(which)					\
 VCL_VOID							\
-VRT_l_##which##_body(VRT_CTX, const char *str, ...)		\
+VRT_l_##which##_body(VRT_CTX, enum lbody_e type, VCL_BLOB blob)	\
 {								\
-	va_list ap;						\
-	const char *p;						\
 	struct vsb *vsb;					\
 								\
+	CHECK_OBJ_NOTNULL(ctx, VRT_CTX_MAGIC);			\
+	AN(blob);						\
 	CAST_OBJ_NOTNULL(vsb, ctx->specific, VSB_MAGIC);	\
-	va_start(ap, str);					\
-	p = str;						\
-	while (p != vrt_magic_string_end) {			\
-		if (p == NULL)					\
-			p = "(null)";				\
-		VSB_cat(vsb, p);				\
-		p = va_arg(ap, const char *);			\
-	}							\
-	va_end(ap);						\
+	if (type == LBODY_SET)					\
+		VSB_clear(vsb);					\
+	assert(type == LBODY_SET || type == LBODY_ADD);		\
+	VSB_bcat(vsb, blob->blob, blob->len);			\
```

Comment:
**later!** not to hold up the PR!

pick up the "array of blobs" (I remember @bsdphk had a good name, but cannot remember which) idea and avoid the copying.